clustpy/deep/dcn.py

```diff
@@ -278,22 +275,22 @@ def _loss(self, batch: list, autoencoder: torch.nn.Module, loss_fn: torch.nn.mod
             ae_loss, embedded, _ = autoencoder.loss(batch, loss_fn, device)
 
         # compute cluster loss
-        assignments = self.predict_hard(embedded)
-        assignment_matrix = int_to_one_hot(assignments, self.centers.shape[0])
-        cluster_loss = self.dcn_loss(embedded, assignment_matrix)
+        labels = self.labels[batch[0]]
```

Comment:
We do not calculate the labels for the loss function but only when minibatch kmeans is executed. Corresponds to original implementation and the paper